Rewrite plugin against Kustomize v3 Plugin API

larslevie opened the pull request:
The Kustomize plugin API is completely different now -- and much more robust.

https://github.com/kubernetes-sigs/kustomize/blob/e34c1ce192d0bff8e0dccf03c9063ccb6d6474be/plugin/doc.go

https://github.com/kubernetes-sigs/kustomize/blob/master/examples/secretGeneratorPlugin.md

Secrets generated from a plugin now require a separate custom resource:

```
apiVersion: kustomize.config.realgeeks.com/v1beta1
kind: SecretsFromVault
metadata:
  name: newman-api-secret
secrets:
  - path: secret/data/prd/am1/kube0/newman-api
    key: google_api
    relabel: foo_google_api
```

larslevie committed:
Rewrite plugin against Kustomize v3 Plugin API

larslevie committed:
Remove unecessary class vars

justinsteffy commented on vault.go:
```diff
@@ -8,92 +8,132 @@ import (
+type VaultSecret struct {
+	Path    string
+	Key     string
+	Relabel string
```
I think `Relabel` is kind of a confusing name. I think either `name` or `secret_name` might be better?

larslevie replied:
The intent was to indicate that this was an optional field only needed when you need to rename the `Key`. It replaces the `@` syntax in the current API. Do you think `SecretName` is clear enough?

justinsteffy replied:
oh...  maybe `SecretKey`? `name` is used by the secret, and `label` has its own meaning in k8s.

justinsteffy commented on vault.go:
```diff
@@ -8,92 +8,132 @@ import (
+		if secret.Relabel != "" {
+			key = secret.Relabel
+		} else {
+			key = secret.Key
```
Didn't this used to be the `{secret_name}_{secret_key}`?

larslevie replied:
Yes, but now that we have a better API, we have the option to have nicer names in `secretKeyRef` blocks. It requires more changes now because we have to touch a bunch of controller manifests, but it makes `secretKeyRefs` a little simpler too.

justinsteffy reviewed (approved):
I few comments, but generally LGTM.

justinsteffy commented:
In my limited testing I've run into some annoyances:

* You have to add the `--enable_alpha_plugins` after the `build`, so it can't be in the entrypoint, unless we add some wrapper script
* Some of my `patchesJson6902` were broken because I had to add `namespace`

larslevie committed:
Change entrypoint: kustomize build + plugins flag

because Kustomize moved the plugins flag to the build command, so it needs to be included here, or it needs to be added to every invocation manually.

larslevie commented:
Moving the flag to `build` from the base `kustomize` was annoying. But the entrypoint could safely become `kustomize build --enable_alpha_plugins` because there aren't any other useful `kustomize` commands -- at least not in the context of how this Docker image is used.
